File: Old VB Version/Postie Notes/PostieMain.vb
Imports System.IO

Public Class PostieMain
    Private Sub NewNoteToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles NewNoteToolStripMenuItem.Click
        Dim postie As New PostieNote
        postie.MdiParent = Me
        postie.Show()
    End Sub

    Private Sub PostieMain_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        For Each arg As String In My.Application.CommandLineArgs
            Dim postie As New PostieNote
            postie.Show()
            postie.RichTextBox1.Text = File.ReadAllText(arg)
        Next

        Dim root As String = "C:\PostieNotes"
        If Not Directory.Exists(root) Then
            Directory.CreateDirectory(root)
        End If

        Environment.GetCommandLineArgs()
    End Sub

    Private Sub OpenNoteToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles OpenNoteToolStripMenuItem.Click
        OpenFileDialog1.InitialDirectory = My.Settings.DefaultSaveDirectory
        OpenFileDialog1.Title = "Open Postie Note"
        OpenFileDialog1.Filter = "Postie Note|*.pns"
        OpenFileDialog1.FileName = ""
        If OpenFileDialog1.ShowDialog = DialogResult.OK Then
            Dim postie As New PostieNote
            postie.Show()
            postie.RichTextBox1.LoadFile(OpenFileDialog1.FileName)
            postie.TextBox1.Text = System.IO.Path.GetFileNameWithoutExtension(OpenFileDialog1.FileName)
        End If
    End Sub

    Private Sub BackupNotesToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles BackupNotesToolStripMenuItem.Click
        Dim FolderDialogue As New FolderBrowserDialog
        FolderDialogue.ShowNewFolderButton = True
        Dim SourcePath As String = "C:\PostieNotes"
        Dim DestinationPath As String
        If (FolderDialogue.ShowDialog() = DialogResult.OK) Then
            DestinationPath = FolderDialogue.SelectedPath
            Dim root As Environment.SpecialFolder = FolderDialogue.RootFolder
            Dim newDirectory As String = System.IO.Path.Combine(DestinationPath, Path.GetFileName(Path.GetDirectoryName(SourcePath)))
            If Not (Directory.Exists(newDirectory)) Then
                Directory.CreateDirectory(newDirectory)
            End If
            Microsoft.VisualBasic.FileIO.FileSystem.CopyDirectory(SourcePath, newDirectory)
        End If
    End Sub

    Private Sub OpenMenuItem_Click(sender As Object, e As EventArgs) Handles OpenMenuItem.Click
        OpenFileDialog1.InitialDirectory = My.Settings.DefaultSaveDirectory
        OpenFileDialog1.Title = "Open Postie Note"
        OpenFileDialog1.Filter = "Postie Note|*.pns"
        OpenFileDialog1.FileName = ""
        If OpenFileDialog1.ShowDialog = DialogResult.OK Then
            Dim postie As New PostieNote
            postie.MdiParent = Me
            postie.Show()
            postie.RichTextBox1.LoadFile(OpenFileDialog1.FileName)
            postie.TextBox1.Text = System.IO.Path.GetFileName(OpenFileDialog1.FileName)
            postie.TextBox1.Text = System.IO.Path.GetFileNameWithoutExtension(OpenFileDialog1.FileName)
        End If
    End Sub

    Private Sub ClearMemoboardToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles ClearMemoboardToolStripMenuItem.Click
        Dim FormNotOpen As Boolean = True
        If MdiChildren.Length > 0 Then
            If MsgBox("Are you sure you want to clear the memoboard? Doing so will close all Postie Notes without saving their contents. Please select 'No' if you want to save your notes.", MsgBoxStyle.YesNo, "Clear Memoboard") = MsgBoxResult.Yes Then
                For Each frm As System.Windows.Forms.Form In Me.MdiChildren
                    frm.Close()
                Next frm
            Else

            End If
        End If
    End Sub

    Private Sub ExitToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles ExitToolStripMenuItem.Click
        End
    End Sub

    Private Sub SettingsToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles SettingsToolStripMenuItem.Click
        PostieSettings.ShowDialog()
    End Sub

    Private Sub PostieMain_FormClosing(sender As Object, e As FormClosingEventArgs) Handles MyBase.FormClosing

    End Sub
End Class

File: Postie Notes/PostieMain.cs
//using System;
//using System.Collections.Generic;
//using System.ComponentModel;
//using System.Data;
//using System.Drawing;
//using System.Text;
using System.Windows.Forms;
using System.IO;

namespace Postie_Notes
{
    public partial class PostieMain : Form
    {
        public PostieMain()
        {
            InitializeComponent();
        }

        private void NewNoteToolStripMenuItem_Click(object sender, System.EventArgs e)
        {
            PostieNote postie = new PostieNote();
            postie.MdiParent = this;
            postie.Show();
        }

        private void PostieMain_Load(object sender, System.EventArgs e)
        {
            //string[] args = ;

            //foreach (string arg in args) 
            //{
            //    PostieNote postie = new PostieNote();
            //    postie.Show();
            //    postie.richTextBoxNote.Text = File.ReadAllText(arg);
            //}

            string root = "C:\\PostieNotes";
            if (!Directory.Exists(root))
            {
                Directory.CreateDirectory(root);
            }

            //System.Environment.GetCommandLineArgs();
        }

        private void OpenMenuItem_Click(object sender, System.EventArgs e)
        {
            openFileDialog1.InitialDirectory = Properties.Settings.Default.DefaultSaveDirectory;
            openFileDialog1.Title = "Open Postie Note";
            openFileDialog1.Filter = "Postie Note|*.pns";
            openFileDialog1.FileName = "";
            if (openFileDialog1.ShowDialog() == DialogResult.OK)
            {
                PostieNote postie = new PostieNote();
                postie.MdiParent = this;
                postie.Show();
                postie.richTextBoxNote.LoadFile(openFileDialog1.FileName);
                postie.textBoxNoteTitle.Text = System.IO.Path.GetFileNameWithoutExtension(openFileDialog1.FileName);
            }
        }

        private void OpenNoteToolStripMenuItem_Click(object sender, System.EventArgs e)
        {
            openFileDialog1.InitialDirectory = Properties.Settings.Default.DefaultSaveDirectory;
            openFileDialog1.Title = "Open Postie Note";
            openFileDialog1.Filter = "Postie Note|*.pns";
            openFileDialog1.FileName = "";
            if (openFileDialog1.ShowDialog() == DialogResult.OK)
            {
                PostieNote postie = new PostieNote();
                postie.Show();
                postie.richTextBoxNote.LoadFile(openFileDialog1.FileName);
                postie.textBoxNoteTitle.Text = System.IO.Path.GetFileNameWithoutExtension(openFileDialog1.FileName);
            }
        }

        private void BackupNotesToolStripMenuItem_Click(object sender, System.EventArgs e)
        {
            FolderBrowserDialog FolderDialogue = new FolderBrowserDialog();
            FolderDialogue.ShowNewFolderButton = true;
            string SourcePath = "C:\\PostieNotes";
            string DestinationPath;
            if (FolderDialogue.ShowDialog() == DialogResult.OK)
            {
                DestinationPath = FolderDialogue.SelectedPath;
                System.Environment.SpecialFolder root = FolderDialogue.RootFolder;
                string newDirectory = System.IO.Path.Combine(DestinationPath, Path.GetFileName(Path.GetDirectoryName(SourcePath)));
                if (!Directory.Exists(newDirectory))
                {
                    Directory.CreateDirectory(newDirectory);
                }
                Microsoft.VisualBasic.FileIO.FileSystem.CopyDirectory(SourcePath, newDirectory);
            }
        }

        private void ExitToolStripMenuItem_Click(object sender, System.EventArgs e)
        {
            Application.Exit();
        }

        private void ClearMemoboardToolStripMenuItem_Click(object sender, System.EventArgs e)
        {
            bool formNotOpen = true;
            if (MdiChildren.Length > 0)
            {
                if (MessageBox.Show("Are you sure you want to clear the memoboard? Doing so will close all Postie Notes without saving their contents. Please select 'No' if you want to save your notes.", "Clear Memoboard", MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.Yes)
                {
                    foreach (Form frm in this.MdiChildren)
                    {
                        frm.Close();
                    }
                }
                else
                {
                    // Nothing.
                }
            }
        }

        private void SettingsToolStripMenuItem_Click(object sender, System.EventArgs e)
        {
            PostieSettings postieSettings = new PostieSettings();
            postieSettings.ShowDialog();
        }
    }
}
